include/zephyr/bluetooth/mesh/msg.h

```diff
@@ -211,6 +213,40 @@ static inline void bt_mesh_msg_ack_ctx_rx(struct bt_mesh_msg_ack_ctx *ack)
 bool bt_mesh_msg_ack_ctx_match(const struct bt_mesh_msg_ack_ctx *ack,
 			       uint32_t op, uint16_t addr, void **user_data);
 
+/**
+ * Message response context.
+ */
+struct bt_mesh_msg_rsp_ctx {
```

Comment:
Not sure if this should be public API or should be hidden.

Reply:
I rethought this and don't see this as a public API for an end application at the moment as it creates unnecessary layering before a message is actually passed to the transport layer. This function helps to reduce the current flash usage, but has branching inside it which prevents the code from being optimized well. I think the common API should be redesigned at some point.